src/main/scala/io/iohk/ethereum/vm/DataWord.scala

```diff
@@ -79,10 +88,56 @@ class DataWord private (private val n: BigInt) extends Ordered[DataWord] {
 
   def *(that: DataWord): DataWord = DataWord(this.n * that.n)
 
-  def /(that: DataWord): DataWord = DataWord(this.n / that.n)
+  def /(that: DataWord): DataWord = zeroCheck(that) { DataWord(this.n / that.n) }
+
+  def sdiv(that: DataWord): DataWord = zeroCheck(that) {
+    DataWord(toUnsignedBigInt(this.signedN / that.signedN))
+  }
+
+  def mod(that: DataWord): DataWord = zeroCheck(that) { DataWord(this.n mod that.n) }
+
+  def smod(that: DataWord): DataWord = zeroCheck(that) {
+    DataWord(this.signedN.signum * (this.signedN.abs mod that.signedN.abs))
+  }
+
+  def addmod(that: DataWord, modulus: DataWord): DataWord = zeroCheck(modulus) {
+    DataWord((this.n + that.n) mod modulus.n)
```

Comment:
It would be a bit faster to use `new DataWord(...)` instead of `DataWord(...)`. Since the resulting integer must be within bounds `[0, MaxValue]` there's no need to call `fixBigInt`.

This also pertains to other methods such as: / (div), sdiv, mod, smod, addmod, mulmod, **, signExtend, getByte, sgt, slt, and, or, xor